### PR TITLE
feat(data-warehouse): planetscale import

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- planetscale requires specific config

## Changes

- create a separate planetscale sync

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/temporal/data_imports/pipelines/sql_database/__init__.py [hunk 1]
```diff
@@ -76,6 +78,13 @@ def sql_source_for_type(
         credentials = ConnectionStringCredentials(
             f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}?ssl_ca={ssl_ca}&ssl_verify_cert=false"
         )
+    elif source_type == ExternalDataSource.Type.PLANETSCALE:
+        is_debug = get_from_env("DEBUG", False, type_cast=str_to_bool)
+        ssl_ca = "/etc/ssl/cert.pem" if is_debug else "/etc/ssl/certs/ca-certificates.crt"
+        credentials = ConnectionStringCredentials(
+            f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}?ssl_ca={ssl_ca}&ssl_verify_cert=false"
+        )
+        connect_args = ["SET workload = 'OLAP';"]
```
Comment (EDsCODE) [inline posthog/temporal/data_imports/pipelines/sql_database/__init__.py:87]: key change

File: posthog/temporal/data_imports/pipelines/sql_database/__init__.py [hunk 2]
```diff
@@ -76,6 +78,13 @@ def sql_source_for_type(
         credentials = ConnectionStringCredentials(
             f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}?ssl_ca={ssl_ca}&ssl_verify_cert=false"
         )
+    elif source_type == ExternalDataSource.Type.PLANETSCALE:
```
Comment (Gilbert09) [inline posthog/temporal/data_imports/pipelines/sql_database/__init__.py:81]: Can I be a massive pain here and suggest we don't add planetscale as a new source? I think it'd be a good idea to make generic sources and not platform-specific (we could end up going down quite a rabbit hole with that) - maintaining multiple versions of a MySQL connector isn't ideal.

We should be able to distinguish planetscale DBs from the hostname - any host ending in `psdb.cloud` is planetscale, we can add the connect args based on that

**Review by Gilbert09 — CHANGES_REQUESTED** [review]

File: posthog/temporal/data_imports/pipelines/sql_database/__init__.py (hunk 2 above)
Comment (EDsCODE) [reply posthog/temporal/data_imports/pipelines/sql_database/__init__.py:81]: good point, missed the bit with psdb!

**Comment by EDsCODE** [comment]
closing in favor of https://github.com/PostHog/posthog/pull/25637

**Pull request closed without merging**